#491 ReadableByteChannel in org.takes.rq.RqMultipart.Base never closes

xupyprmv opened the pull request:
#491. I added `.close()` call in finally block. Also I removed two checkstyle annotations that are not actual.

davvd commented:
@xupyprmv Thanks, let me find someone who can review this pull request

davvd commented:
@dmzaytsev please review, thanks

dmzaytsev commented:
@xupyprmv looks good, thanks

dmzaytsev commented:
@rultor merge

rultor commented:
> @rultor merge

@dmzaytsev Thanks for your request. @yegor256 Please confirm this.

yegor256 commented on src/main/java/org/takes/rq/RqMultipart.java:
```diff
@@ -132,12 +130,20 @@
+                );
+                this.map = this.buildRequests(req);
+            } finally {
+                if (this.body == null) {
```
@xupyprmv `null` is an anti-pattern

yegor256 commented:
@xupyprmv what you're doing with this `if body == null` is a very bad approach. you need two try/catch blocks instead.

xupyprmv commented:
@yegor256 I added changes according to your remark. Please, continue with CR.

yegor256 commented:
@xupyprmv looks correct now, but what about a unit test? How do you know that the code didn't work before your changes and works correctly after them?

xupyprmv commented:
@yegor256 I added unit test. It not passes with old version and passes with new version.
I used not best approach (reflection) to create test, but it is better then open access to `body` field.

Please, continue with CR.

xupyprmv commented:
@yegor256 Please, continue with CR.

yegor256 commented:
@xupyprmv oh, no... it's a terrible idea to do what you're doing in this test. the idea of a unit test is to help me to refactor my class without touching the test. your tests works in an opposite direction. never ever use reflection in unit testing. you're allowed to test only public interfaces of classes.

xupyprmv commented:
@yegor256 

> looks correct now, but what about a unit test?

...

> oh, no... it's a terrible idea 

There is no other way to test private field without public access.

> you're allowed to test only public interfaces of classes 

So, I deleted test. Bug mentioned in #491 cannot be reproduced with test, that operates only with public interface. Also, fixed code cannot be approved by test, that operates only with public interface.

@yegor256 Please continue with CR.

yegor256 commented:
@rultor merge

rultor commented:
> @rultor merge

@yegor256 OK, I'll try to merge now. You can check the progress of the merge [here](http://www.rultor.com/t/6384-174804181)

rultor commented:
> @rultor merge

@yegor256 Done! FYI, the full log is [here](http://www.rultor.com/t/6384-174804181) (took me 17min)

davvd commented:
@elenavolokhova please, check this issue for QA compliance, as per [par.24](http://at.teamed.io/policy.html#24)

davvd commented:
@rultor deploy

rultor commented:
> @rultor deploy

@davvd OK, I'll try to deploy now. You can check the progress [here](http://www.rultor.com/t/6384-175582363)

rultor commented:
> @rultor deploy

@davvd Done! FYI, the full log is [here](http://www.rultor.com/t/6384-175582363) (took me 12min)

elenavolokhova commented:
@dmzaytsev 
According to our [quality rules](http://www.teamed.io/qa.html):
 > The code reviewer found at least three problems in the code.
 
 There were no issues found during review.
Please confirm that you will try to find at least three problems next time.

dmzaytsev commented:
@elenavolokhova I confirm

elenavolokhova commented:
@davvd Quality is acceptable here.

davvd commented:
> @davvd Quality is acceptable here.

@elenavolokhova thank you, let's try to make it "good" next time

davvd commented:
@dmzaytsev paied 10 mins to @elenavolokhova for QA review (payment ID is `75810218`)... **18 mins** added to your account (payment number `75810260`), many thanks for your contribution! 254 hours and 5 mins spent here.... extra minutes for review comments (c=3)... added +18 to your rating, now it is equal to [+2834](http://www.netbout.com/b/35166?open=rating)
